### PR TITLE
Eth keys

### PR DESCRIPTION
**Pull request by VoR0220**
Ethereum keys separated so that we don't run into the damn symbolization error anymore! :D

### CODE REVIEW

File: eris-keys/server.go [hunk 1]
```diff
@@ -199,8 +206,10 @@ func verifyHandler(w http.ResponseWriter, r *http.Request) {
 		WriteError(w, fmt.Errorf("must provide a signature with the `sig` key"))
 		return
 	}
-
-	res, err := coreVerify(typ, pub, msg, sig)
+	Pub, _ := hex.DecodeString(pub)
+	Msg, _ := hex.DecodeString(msg)
+	Sig, _ := hex.DecodeString(sig)
+	res, err := coreVerify(typ, Pub, Msg, Sig)
```
Comment (pietv) [inline eris-keys/server.go]: Let's be more defensive about this (checking for `err`)? It looks better and avoids sudden panics (or worse: unexpected empty strings).

File: eris-keys/core.go [hunk 2]
```diff
@@ -201,17 +202,21 @@ func coreVerify(typ string, pub, hash, sig []byte) (result bool, err error) {
...
-	return result, nil
+	return
 }
 
-func corePub(addr []byte) ([]byte, error) {
-	key, err := GetKey(addr)
+func corePub(addr string) ([]byte, error) {
+	addrB, err := hex.DecodeString(addr)
+	if err != nil {
+		return nil, fmt.Errorf("addr is invalid hex: %s", err.Error())
```
Comment (pietv) [inline eris-keys/core.go]: `err.Error()` is stringizer, `Errorf` can call it for you: `fmt.Errorf("addr is invalid hex: %v", err)`, but that's a matter of taste

File: crypto/key.go [hunk 3]
```diff
@@ -309,18 +306,13 @@ func IsValidKeyJson(j []byte) []byte {
...
-	/*if err != nil {
-
-		return nil, fmt.Errorf("Pub key problems: %v", err)
-	}*/
 	id, _ := uuid.NewRandom()
 	return &Key{
 		Id:         id,
 		Type:       KeyType{CurveTypeSecp256k1, addrType},
-		Address:    AddressFromPub(addrType, pub),
+		Addr:    	AddressFromPub(addrType, ethCrypto.FromECDSAPub(priv.Public().(*ecdsa.PublicKey))),
```
Comment (pietv) [inline crypto/key.go]: `gofmt` (returns some diffs on `EthKeys` branch)

File: crypto/key.go [hunk 4]
```diff
@@ -332,10 +332,10 @@ func newKeyEd25519(addrType AddrType) *Key {
 }
 
 func keyFromPrivSecp256k1(addrType AddrType, priv []byte) (*Key, error) {
-	pub, err := secp256k1.GeneratePubKey(priv)
-	if err != nil {
-		return nil, err
-	}
+	pub := []byte{'1'}//secp256k1.GeneratePubKey(priv)
+	//if err != nil {
```
Comment (pietv) [inline crypto/key.go]: garbage. maybe drop all commented code? especially `/* ... */`

File: crypto/helpers/helpers.go [hunk 5]
```diff
@@ -52,6 +53,28 @@ func (pubKey PubKeyEd25519) Address() []byte {
...
+}
+
+type (
+	Address [20]byte
+)
+
+func (a Address) Bytes() []byte { return a[:] }
 
+func (a *Address) SetBytes(b []byte) {
+	if len(b) > len(a) {
```
Comment (pietv) [inline crypto/helpers/helpers.go]: Didn't dig into the intention of this function, but `b` is inserted not to the end of `a` but at `a`-1 (leaving the last symbol of `a`, hope this was intended). I think deserves a comment. Also, if `len(a)` is less than 20 the method will panic.

File: crypto/key.go (hunk 4 above)
Comment (VoR0220) [inline crypto/key.go]: shit...I need to get rid of that.

**Comment by pietv** [comment]
LGTM

File: crypto/helpers/helpers.go (hunk 5 above)
Comment (pietv) [inline crypto/helpers/helpers.go]: Cool.

File: eris-keys/server.go (hunk 1 above)
Comment (VoR0220) [inline eris-keys/server.go]: done

**Pull request merged**